src/lib/components/ui/dialog/dialog-content.svelte

```diff
@@ -21,45 +21,53 @@
 	export let hide_close_button: $$Props['hide_close_button'] = false;
 	export let overlay_classes: $$Props['overlay_classes'] = undefined;
 	export let el: $$Props['el'] = undefined;
+
+	$: _transition = transition || flyAndScale;
 </script>
 
 <Dialog.Portal>
 	<Dialog.Overlay class={overlay_classes} />
-	<DialogPrimitive.Content
-		bind:el
-		{transition}
-		{transitionConfig}
-		class={cn(
-			'fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background-100 p-6 shadow-lg sm:rounded-lg md:w-full',
-			className
-		)}
-		{...$$restProps}
-	>
-		<slot />
-		{#if !hide_close_button}
-			<DialogPrimitive.Close asChild let:builder>
-				{#if close_button === 'esc'}
-					<Button
-						builders={[builder]}
-						size="sm"
-						variant="secondary"
-						class="absolute right-4 top-3 h-5 px-1.5 text-xs "
-					>
-						Esc
-					</Button>
-				{:else if close_button === 'x'}
-					<Button
-						svg_only
-						aria-label="Close"
-						shape="square"
-						size="tiny"
-						variant="secondary"
-						class="absolute right-4 top-3 px-1.5"
-					>
-						<Icons.Cross aria-hidden="true" class="size-5" />
-					</Button>
+	<DialogPrimitive.Content asChild let:builder>
+		<div class="fixed inset-0 z-50 grid place-items-center">
+			<div
+				bind:this={el}
+				transition:_transition={transitionConfig}
+				use:builderActions={{ builders: [builder] }}
+				{...getAttrs([builder])}
+				{...$$restProps}
+				class={cn(
+					'fixed left-[50%] top-[50%] z-50 grid w-full max-w-lg translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background-100 p-6 shadow-lg sm:rounded-lg md:w-full',
+					className
```

Comment:
I think the implementation is wrong. I believe using translate was one of the contributing factors.

Reply:
Snap. I have forgotten to remove those class names. Should be fixed now.

Reply:
Noticed the `x` issue. Pushing a fix now.